packages/salesforce/src/Adaptor.js

```diff
@@ -363,17 +364,37 @@ export function bulkQuery(qs, options, callback) {
  * Create and execute a bulk job.
  * @public
  * @example
- * bulk('Patient__c', 'insert', { failOnError: true, pollInterval: 3000, pollTimeout: 240000 }, state => {
- *   return state.data.someArray.map(x => {
- *     return { 'Age__c': x.age, 'Name': x.name }
- *   })
- * });
+ * bulk(
+ *   "Patient__c",
+ *   "insert",
+ *   { failOnError: true, pollInterval: 3000, pollTimeout: 240000 },
+ *   (state) => state.someArray.map((x) => ({ Age__c: x.age, Name: x.name }))
+ * );
+ * @example <caption>Bulk upsert</caption>
+ * bulk(
+ *   "vera__Beneficiary__c",
+ *   "upsert",
+ *   { extIdField: "vera__External_ID__c", failOnError: true },
+ *   [
+ *     {
+ *       vera__Reporting_Period__c: 2023,
+ *       vera__Geographic_Area__c: "Uganda",
+ *       "vera__Indicator__r.vera__ExtId__c": 1001,
+ *       vera__Result_UID__c: "1001_2023_Uganda",
+ *     },
+ *   ]
+ * );
  * @function
- * @param {String} sObject - API name of the sObject.
- * @param {String} operation - The bulk operation to be performed
- * @param {Object} options - Options passed to the bulk api.
- * @param {Function} records - an array of records, or a function which returns an array.
- * @returns {Operation}
+ * @param {string} sObject - API name of the sObject.
+ * @param {string} operation - The bulk operation to be performed.Eg "insert" | "update" | "upsert"
+ * @param {object} options - Options passed to the bulk api.
+ * @param {integer} [options.pollTimeout=240000] - Polling timeout in milliseconds.
+ * @param {integer} [options.pollInterval=6000] - Polling interval in milliseconds.
+ * @param {string} [options.extIdField] - External id field.
+ * @param {boolean} [options.failOnError] - Fail the operation on error.
+ * @param {boolean} [options.allowNoOp] - For skipping operation.
```

Comment:
From what i have understood `allowNoOp` is basically for skipping the operation if we don't have any records for the operation. This smells like a problem that can be solved with `fnIf()`

Reply:
Right, so we send that flag to salesforce and salesforce won't throw an error if we send an empty array?

I agree that fnif is a better solution.

Let's remove this documentation - we never documented this option before and, if you know the option from jsforce, it will continue to work. But let's not call attention to it.

Reply:
It's not a jsforce thingy or salesforce option. It's specific for this adaptor function. 
If i am to translate it using `fnIf` it will be something like this 
```
fnIf(
  $.records.length > 0,
  bulk(
    "vera__Beneficiary__c",
    "upsert",
    { extIdField: "vera__Result_UID__c" },
    $.records
  )
);
```

Reply:
@josephjclark removed `allowNoOp` docs